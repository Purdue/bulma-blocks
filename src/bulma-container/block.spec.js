jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    insertBlock
} from '@wordpress/e2e-test-utils';

import {
    clickElementByText,
    blockStartup,
    clickCheckbox,
    updateRangeInput, 
    selectOption
} from '../test-helpers'

const block = {blockTitle: 'Bulma - Container', blockName: 'bulma-blocks/container'}

describe( `ðŸ”¬ ${block.blockTitle} Block`, () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'ðŸ”Ž Block should be available.', async () => {
        await insertBlock(block.blockTitle)

        // tests that the block is properly inserted and matches the existing snapshot
        expect(await page.$(`[data-type="${block.blockName}"]`)).not.toBeNull()
        expect( await getEditedPostContent() ).toMatchSnapshot()
    } )

    describe( 'ðŸ”¬ Side Panel Settings', () => {
        describe( 'ðŸ”¬ Checkbox Settings', () => {
            test( 'ðŸ”Ž Full Width Option', async () => {
                await blockStartup(block)

                await clickCheckbox('Fluid Full-width')
                
                const editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"fluidChecked":true')).toBe(true)
                expect( editedContent ).toMatchSnapshot()
            })
        })
        describe( 'ðŸ”¬ Dropdown Settings', () => {
            test( 'ðŸ”Ž Background Color Options', async () => {
                await blockStartup(block)

                // black background option select
                await selectOption('Background Color', 'has-background-black')
                
                let editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"bgColor":"has-background-black"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()

                
                // light grey background option select
                await selectOption('Background Color', 'has-background-grey-lighter')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"bgColor":"has-background-grey-lighter"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()
                

                // dark grey background option select
                await selectOption('Background Color', 'has-background-black-ter')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"bgColor":"has-background-black-ter"')).toBe(true)
                expect( editedContent ).toMatchSnapshot()


                // no background option select (DEFAULT)
                await selectOption('Background Color', '')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent.includes('"bgColor":"has-background-black"')).toBe(false)
                expect( editedContent.includes('"bgColor":"has-background-grey-lighter"')).toBe(false)
                expect( editedContent.includes('"bgColor":"has-background-black-ter"')).toBe(false)
                expect( editedContent ).toMatchSnapshot()
            })

        })
    })

})