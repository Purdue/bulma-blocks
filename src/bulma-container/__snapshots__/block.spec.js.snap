// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🔬 Bulma - Container Block 🔎 Block should be available. 1`] = `
"<!-- wp:bulma-blocks/container -->
<div class=\\"container  \\"></div>
<!-- /wp:bulma-blocks/container -->"
`;

exports[`🔬 Bulma - Container Block 🔬 Side Panel Settings 🔬 Checkbox Settings 🔎 Full Width Option 1`] = `
"<!-- wp:bulma-blocks/container {\\"fluidChecked\\":true} -->
<div class=\\"container  is-fluid \\"></div>
<!-- /wp:bulma-blocks/container -->"
`;

exports[`🔬 Bulma - Container Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Background Color Options 1`] = `
"<!-- wp:bulma-blocks/container {\\"bgColor\\":\\"has-background-black\\"} -->
<div class=\\"container  has-background-black\\"></div>
<!-- /wp:bulma-blocks/container -->"
`;

exports[`🔬 Bulma - Container Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Background Color Options 2`] = `
"<!-- wp:bulma-blocks/container {\\"bgColor\\":\\"has-background-grey-lighter\\"} -->
<div class=\\"container  has-background-grey-lighter\\"></div>
<!-- /wp:bulma-blocks/container -->"
`;

exports[`🔬 Bulma - Container Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Background Color Options 3`] = `
"<!-- wp:bulma-blocks/container {\\"bgColor\\":\\"has-background-black-ter\\"} -->
<div class=\\"container  has-background-black-ter\\"></div>
<!-- /wp:bulma-blocks/container -->"
`;

exports[`🔬 Bulma - Container Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Background Color Options 4`] = `
"<!-- wp:bulma-blocks/container -->
<div class=\\"container  \\"></div>
<!-- /wp:bulma-blocks/container -->"
`;
